kernel_gateway/jupyter_websocket/__init__.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+"""Jupyter websocket personality for the Kernel Gateway"""
+
+from ..base.handlers import default_handlers as default_base_handlers
+from ..services.activity.handlers import ActivityHandler
+from ..services.kernels.pool import KernelPool
+from ..services.kernels.handlers import default_handlers as default_kernel_handlers
+from ..services.kernelspecs.handlers import default_handlers as default_kernelspec_handlers
+from ..services.sessions.handlers import default_handlers as default_session_handlers
+from .handlers import default_handlers as default_api_handlers
+from notebook.utils import url_path_join
+
+class JupyterWebsocketPersonality(object):
+    """Personality for standard websocket functionality, registering
+    endpoints that are part of the Jupyter Kernel Gateway API
+    """
+    def __init__(self, gatewayapp):
+        pass
+
+    def init_configurables(self, gatewayapp):
```

Comment:
This API is looking better. One more comment on consistency. The constructor gets a gatewayapp instance. `init_configurables` also gets it. The other methods do not.

How about just passing it to the constructor and letting the instance save a ref?

This is a bit of forward planning too. We'll want to make these LoggingConfigurables so that we can use traitlets to configure them. The convention for passing a ref like app through is to use the `parent` argument. Maybe we should just inherit now and adopt that convention so we don't have to change API when we switch over. It should be simple:

1. Have the personality classes inherit from LoggingConfigurable. 
2. Pass the gatewayapp instance to the personality constructor via a `parent=gatewayapp` keyword arg.
3. Have the constructor of the personality class call the ancestor constructors passing *args and **kwargs.
4. Change any code that needs the app ref to use `self.parent`.

Have a look at how ActivityManager's base class and constructor are definedhttps://github.com/jupyter/kernel_gateway/blob/master/kernel_gateway/services/activity/manager.py#L44 and how an instance is instantiated https://github.com/jupyter/kernel_gateway/blob/master/kernel_gateway/gatewayapp.py#L297. (Might as well pass the `log` too while you're at it.)